rskj-core/src/main/java/co/rsk/peg/BridgeSerializationUtils.java

```diff
@@ -61,46 +61,77 @@ private BridgeSerializationUtils() {
         throw new IllegalAccessError("Utility class, do not instantiate it");
     }
 
+    public static byte[] serializeSingleEntry(Map.Entry<Keccak256, BtcTransaction> entry) {
+        byte[][] entryBytes = serializeMapEntry(entry);
+        return RLP.encodeList(entryBytes);
+    }
+
     public static byte[] serializeMap(SortedMap<Keccak256, BtcTransaction> map) {
         int ntxs = map.size();
-
         byte[][] bytes = new byte[ntxs * 2][];
         int n = 0;
 
         for (Map.Entry<Keccak256, BtcTransaction> entry : map.entrySet()) {
-            bytes[n++] = RLP.encodeElement(entry.getKey().getBytes());
-            bytes[n++] = RLP.encodeElement(entry.getValue().bitcoinSerialize());
+            byte[][] entryBytes = serializeMapEntry(entry);
+            bytes[n++] = entryBytes[0];
+            bytes[n++] = entryBytes[1];
         }
 
         return RLP.encodeList(bytes);
     }
 
-    public static SortedMap<Keccak256, BtcTransaction> deserializeMap(byte[] data, NetworkParameters networkParameters, boolean noInputsTxs) {
+    private static byte[][] serializeMapEntry(Map.Entry<Keccak256, BtcTransaction> entry) {
```

Comment:
wdyt about having two specific and separated methods, `serializeMapKey` and `serializeMapValue`, since this method is actually doing that and not serializing the whole map?

it would imply having this:
```
            bytes[n++] = serializeMapKey(entry);
            bytes[n++] = serializeMapValue(entry);
```
            
instead of this:
```
            byte[][] entryBytes = serializeMapEntry(entry);
            bytes[n++] = entryBytes[0];
            bytes[n++] = entryBytes[1];
```

which i find more descriptive and easier to read :)

Reply:
I understand the reasoning, but I think this would be confusing since in RLP format a key-value pair is encoded as a 2d byte array. Also consider we would have to be doing the same for the `serializeSingleEntry` method which would look a bit off. wdyt?

Reply:
good point, lets keep it that way. Thanks!